acmepos-ui/MainUnit.pas

```diff
@@ -500,17 +504,25 @@ procedure SecretsChanged(e: SPIClient_TLB.Secrets); stdcall;
 procedure SpiStatusChanged(e: SPIClient_TLB.SpiStatusEventArgs); stdcall;
 begin
   if (not Assigned(frmActions)) then
-  begin
-    frmActions := frmActions.Create(frmMain, Spi);
-    frmActions.PopupParent := frmMain;
-    frmMain.Enabled := False;
-  end;
-
-  frmActions.Show;
-
-  if (Spi.CurrentFlow = SpiFlow_Idle) then
-    frmActions.richEdtFlow.Lines.Clear();
-
+  begin
+    if (not Init) then
+    begin
+      frmActions := TfrmActions.Create(frmMain, Spi);
+      frmActions.PopupParent := frmMain;
+      frmMain.Enabled := False;
+    end;
+  end;
+
+  if (not Init) then
+  begin
+    frmActions.Show;
+
+    if (Spi.CurrentFlow = SpiFlow_Idle) then
+      frmActions.richEdtFlow.Lines.Clear();
+
+    Init := False;

```

Comment:
Isn't `Init` already false if you got into this `then` block?

Reply:
I can see that. My point is that `Init := False` only gets executed if `not Init` condition evaluates to true (meaning that `Init` is already `False`), right?

Reply:
I mixed the place of the code. you're right. there is no need